pkg/backends/newrelic/newrelic.go

```diff
@@ -407,13 +445,20 @@ func (n *Client) postWrapper(ctx context.Context, json []byte, dataType string)
 		if err != nil {
 			return fmt.Errorf("error POSTing: %s", err.Error())
 		}
+
+		if resp.StatusCode == http.StatusTooManyRequests {
+			return &RetryAfterError{Response: resp}
```

Comment:
This block should be moved in to the `if` block below, so that the `defer` and `ReadAll` are both handled - if you don't properly close a response body (ie, read all the data and call Close), then you don't get connection reuse.

You could put this check between the `ReadAll` and `logger.Info`, that way we do the read, but don't log when we know it's not a problem.

- [ ] properly close response body

Reply:
I've addressed this in last commit.